### PR TITLE
refactor: clean long division code

### PR DESCRIPTION
**Pull request by saahil-mahato**
## Description
There was a redundant code block in `LongDivision.java`. This PR simply deletes it and does a few minor code quality modifications.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/maths/LongDivisionTest.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 import org.junit.jupiter.api.Test;
 
-public class LongDivisionTest {
+class LongDivisionTest {
```
Comment (alxkm) [inline src/test/java/com/thealgorithms/maths/LongDivisionTest.java:7]: What the benefits to remove `public`?

**Review by alxkm — CHANGES_REQUESTED** [review]

File: src/test/java/com/thealgorithms/maths/LongDivisionTest.java (hunk 1 above)
Comment (saahil-mahato) [reply src/test/java/com/thealgorithms/maths/LongDivisionTest.java:7]: This was a code smell issue sonarlint. Starting from Junit 5 classes do not need to be public.

File: src/main/java/com/thealgorithms/maths/LongDivision.java [hunk 2]
```diff
@@ -39,14 +39,7 @@ public static int divide(int dividend, int divisor) {
...
             long part1 = Long.parseLong(partV1);
-            if (part1 > newDivisor1) {
-                int quotient = 0;
-                while (part1 >= newDivisor1) {
-                    part1 = part1 - newDivisor1;
-                    quotient++;
-                }
-                answer.append(quotient);
-            } else if (part1 == newDivisor1) {
+            if (part1 >= newDivisor1) {
```
Comment (alxkm) [inline src/main/java/com/thealgorithms/maths/LongDivision.java:42]: Why did you remove it? Maybe it wasn't covered by tests? Because it seems to be inefficient but requires code. 

Perhaps this could be written more efficiently, like this:

```
    int quotient = (int) (part1 / newDivisor1);
    part1 = part1 % newDivisor1;
    answer.append(quotient);
```

**Review by alxkm — CHANGES_REQUESTED** [review]
